web/src/components/tasks/anytime-action/tax-clearance-certificate/TaxClearanceStepThree.tsx

```diff
@@ -48,8 +48,15 @@ export const TaxClearanceStepThree = (props: Props): ReactElement => {
   const handleButtonClick = async (): Promise<void> => {
     if (!userData) return;
     const taxClearanceResponse = await api.postTaxClearanceCertificate(userData);
-    if (taxClearanceResponse.certificatePdfArray)
-      props.setCertificatePdfArray(taxClearanceResponse.certificatePdfArray);
+    if (taxClearanceResponse.certificatePdfArray) {
```

Comment:
[sand] Should we check for length here too?

```typescript
if (taxClearanceResponse.certificatePdfArray && taxClearanceResponse.certificatePdfArray.length > 0) {
```

Reply:
Yes, but we will need corresponding logs to help us diagnose the issue, which means we'd need to check within `api` and log it using `logger` so we can catch it in CloudWatch and then handle the error like others by alarming or by sending a Slack message.

In the frontend, this would catch a server communication error between us and our backend, which is much less likely and might warrant consideration of an error boundary, which we don't have a ton of precedent in our code (but I'd like to see us adopt).

So, yeah. Imagine I made this comment on our backend, since we don't have a frontend observability framework.